File: tasks/704.py
# 704. Binary Search
import math


class Solution:
    def search(self, nums: list[int], target: int) -> int:
        s = 0
        e = len(nums)
        iter = math.log2(e)
        print(iter)
        for i in range(0, int(math.floor(math.log2(e))) + 1):
            m = (s + e) // 2
            if nums[m] == target:
                return m
            elif nums[m] > target:
                e = m
            else:
                s = m
        return -1

File: solutions/704/704.go
package _704

func search(nums []int, target int) int {
	a := 0
	b := len(nums)

	for a != b {
		c := (a + b) / 2

		value := nums[c]
		if value == target {
			return c
		}
		if c == a {
			break
		}

		if value > target {
			b = c
		} else {
			a = c
		}
	}

	return -1
}
